internal/page/donor/upload_evidence.go

```diff
@@ -51,29 +72,33 @@ func UploadEvidence(tmpl template.Template, payer Payer, donorStore DonorStore,
 			data.Errors = form.Validate()
 
 			if data.Errors.None() {
-				for _, file := range form.Files {
-					uuid := randomUUID()
-					key := lpa.UID + "-evidence-" + uuid
-
-					_, err := s3Client.PutObject(r.Context(), &s3.PutObjectInput{
-						Bucket:               aws.String(evidenceBucketName),
-						Key:                  aws.String(key),
-						Body:                 bytes.NewReader(file.Data),
-						ServerSideEncryption: types.ServerSideEncryptionAwsKms,
-						Tagging:              aws.String("replicate=true"),
-					})
-					if err != nil {
-						return err
+				if form.Action == "upload" {
+					for _, file := range form.Files {
+						uuid := randomUUID()
+						key := lpa.UID + "-evidence-" + uuid
+
+						_, err := s3Client.PutObject(r.Context(), &s3.PutObjectInput{
+							Bucket:               aws.String(evidenceBucketName),
+							Key:                  aws.String(key),
+							Body:                 bytes.NewReader(file.Data),
+							ServerSideEncryption: types.ServerSideEncryptionAwsKms,
+							Tagging:              aws.String("replicate=true"),
+						})
+						if err != nil {
+							return err
+						}
+
+						lpa.EvidenceKeys = append(lpa.EvidenceKeys, page.Evidence{Key: key, Filename: file.Filename})
```

Comment:
Is there any filename normalisation going on in `form.Files` that I've missed? Otherwise I can add some XSS in a filename to pass downstream.